sdk/typespec/typespec_client_core/src/http/response.rs

```diff
@@ -146,16 +150,60 @@ impl<T> Response<T> {
     /// #    parse_response(r).await;
     /// # }
     /// ```
-    pub async fn deserialize_body_into<U: Model>(self) -> crate::Result<U> {
-        U::from_response_body(self.body).await
+    #[cfg(feature = "json")]
+    pub async fn into_json_body<U: DeserializeOwned>(self) -> crate::Result<U> {
+        self.into_raw_body().json().await
+    }
+
+    /// Fetches the entire body and tries to deserialize it as XML into type `U`.
+    ///
+    /// This method is intended for situations where:
+    ///
+    /// 1. The response was not given a specific type by the service SDK (i.e. the API returned `Response<ResponseBody>`, indicating a raw response).
+    /// 2. You want to deserialize the response into your OWN type, rather than the default type specified by the service SDK.
+    ///
+    /// # Example
+    /// ```rust
+    /// # pub struct GetSecretResponse { }
+    /// use typespec_client_core::http::{Model, Response};
+    /// # #[cfg(not(feature = "derive"))]
+    /// # use typespec_macros::Model;
+    /// use serde::Deserialize;
+    /// use bytes::Bytes;
+    ///
+    /// #[derive(Model, Deserialize)]
+    /// struct MySecretResponse {
+    ///    value: String,
+    /// }
+    ///
+    /// async fn parse_response(response: Response<GetSecretResponse>) {
+    ///   // Calling `into_xml_body` will parse the body into `MySecretResponse` instead of `GetSecretResponse`.
+    ///   let my_struct: MySecretResponse = response.into_xml_body().await.unwrap();
+    ///   assert_eq!("hunter2", my_struct.value);
+    /// }
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() {
+    /// #    let r: Response<GetSecretResponse> = typespec_client_core::http::Response::from_bytes(
+    /// #      http_types::StatusCode::Ok,
+    /// #      typespec_client_core::http::headers::Headers::new(),
+    /// #      "<Response><name>database_password</name><value>hunter2</value></Response>",
```

Comment:
FYI, security scanners might flag this as a secret... it looks real enough even though we all know it isn't

Reply:
Good point. We already had a JSON equivalent to this and it hasn't been caught so far, but we can watch out for it.

Reply:
Could use "hunter2". That'll get masked automatically, right?